spec/features/admin/order_cycles_complex_nav_check_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+feature '
+    As an administrator
+    I want to be alerted when I navigate away from a dirty order cycle form
+', js: true do
+  include AuthenticationWorkflow
+
+  scenario "alert when navigating away from dirty form" do
+    # Given an order cycle with all the settings
+    oc = create(:order_cycle)
+    oc.suppliers.first.update_attribute :name, 'AAA'
+    oc.suppliers.last.update_attribute :name, 'ZZZ'
+    oc.distributors.first.update_attribute :name, 'AAAA'
+    oc.distributors.last.update_attribute :name, 'ZZZZ'
```

Comment:
I wonder why do we need these updates?
FYI you could use simple_order_cycle factory like create(:simple_order_cycle) if that would reduce number of lines here.

Reply:
Yeah good point, my copy pasta is showing. Removed in 223521d

From what I've learnt I think it needs `create(:order_cycle)` over `create(:simple_order_cycle)` to get the complex form variant though.